pkg/logs/launchers/integration/launcher_test.go

```diff
@@ -385,20 +386,28 @@ func (suite *LauncherTestSuite) TestSentLogExceedsTotalUsage() {
 	suite.s.combinedUsageMax = 3 * 1024 * 1024
 
 	// Given 3 files exist
-	filename1 := "sample_integration1_123.log"
-	filename2 := "sample_integration2_123.log"
-	filename3 := "sample_integration3_123.log"
-	files := [3]string{filename1, filename2, filename3}
+	fileWithPath1 := filepath.Join(suite.s.runPath, "sample_integration1_123.log")
+	fileWithPath2 := filepath.Join(suite.s.runPath, "sample_integration2_123.log")
+	fileWithPath3 := filepath.Join(suite.s.runPath, "sample_integration3_123.log")
+	fileNames := [3]string{fileWithPath1, fileWithPath2, fileWithPath3}
 
 	//  And I write 1Mb to each file in seq order
 	dataOneMB := make([]byte, 1*1024*1024)
-	for _, filename := range files {
-		file, err := os.Create(filepath.Join(suite.s.runPath, filename))
+	for _, fileWithPath := range fileNames {
+		file, err := os.Create(fileWithPath)
 		require.NoError(suite.T(), err)
 		_, _ = file.Write(dataOneMB)
 		_ = file.Close()
 	}
 
+	// If the files have the same timestamp, scanInitialFiles will detect them in
+	// random order. Setting their modified time manually allows the
+	// scanInitialFiles function to detect them in a deterministic manner
+	modTime := time.Now()
+	accessTime := time.Now()
+	os.Chtimes(fileWithPath2, accessTime, modTime)
```

Comment:
Do we also want to set the mod time for file `1`? If this code runs fast enough couldn't file 1 and 2 end up with the same timestamp?

Reply:
Oh true. It hasn't flaked for file 1 yet since it gets written to, but it definitely seems possible that it could run fast enough that file 1 would have the same timestamp as file 2.

Changing it now